pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,654 @@
+	vmeInfo.Insecure = insecure
+	vmeInfo.KeepVme = keepVme
+	vmeInfo.VolumeName = volumeName
+
+	return nil
+}
+
+// getVirtualMachineExport serves as a wrapper to get the VirtualMachineExport object
+func getVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExportInfo) (*exportv1.VirtualMachineExport, error) {
+	vmexport, err := client.VirtualMachineExport(vmeInfo.Namespace).Get(context.TODO(), vmeInfo.Name, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return vmexport, nil
+}
+
+// CreateVirtualMachineExport serves as a wrapper to create the virtualMachineExport object and, if needed, do error handling
+func CreateVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExportInfo) error {
+	vmexport, err := getVirtualMachineExport(client, vmeInfo)
+	if err != nil {
+		return err
+	}
+	if vmexport != nil {
+		return fmt.Errorf("VirtualMachineExport '%s/%s' already exists", vmeInfo.Namespace, vmeInfo.Name)
+	}
+
+	vmexport = &exportv1.VirtualMachineExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vmeInfo.Name,
+			Namespace: vmeInfo.Namespace,
+		},
+		Spec: exportv1.VirtualMachineExportSpec{
+			TokenSecretRef: getExportSecretName(vmeInfo.Name),
+			Source:         vmeInfo.ExportSource,
+		},
+	}
+
+	vmexport, err = client.VirtualMachineExport(vmeInfo.Namespace).Create(context.TODO(), vmexport, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Generate/get secret to be used with the vmexport
+	_, err = getOrCreateTokenSecret(client, vmexport)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' created succesfully\n", vmeInfo.Namespace, vmeInfo.Name)
+	return nil
+}
+
+// DeleteVirtualMachineExport serves as a wrapper to delete the virtualMachineExport object
+func DeleteVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExportInfo) error {
+	if err := client.VirtualMachineExport(vmeInfo.Namespace).Delete(context.TODO(), vmeInfo.Name, metav1.DeleteOptions{}); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		fmt.Printf("VirtualMachineExport '%s/%s' does not exist", vmeInfo.Namespace, vmeInfo.Name)
+		return nil
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' deleted succesfully\n", vmeInfo.Namespace, vmeInfo.Name)
+	return nil
+}
+
+// DownloadVirtualMachineExport handles the process of downloading the requested volume from a VirtualMachineExport object
+func DownloadVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExportInfo) error {
+	if vmeInfo.ShouldCreate {
+		if err := CreateVirtualMachineExport(client, vmeInfo); err != nil {
+			if !errExportAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+
+	if !vmeInfo.KeepVme {
+		defer DeleteVirtualMachineExport(client, vmeInfo)
+	}
+
+	// Wait for the vmexport object to be ready
+	if err := ExportProcessingComplete(client, vmeInfo, processingWaitInterval, processingWaitTotal); err != nil {
```

Comment:
suggestion: make the function return the completed vmexport instead of calling the same get function again in next line?

Reply:
The problem there is that the `Get` is used inside `PollImmediate`, wich only returns error, so I'd need another Get call anyway. Being that the case, I think I prefer having it outside `ExportProcessingComplete`.